cmd/simulator/worker/worker.go

```diff
@@ -0,0 +1,321 @@
+// (c) 2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package worker
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/big"
+	"math/rand"
+	"time"
+
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/params"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/ava-labs/subnet-evm/cmd/simulator/key"
+	"github.com/ava-labs/subnet-evm/cmd/simulator/metrics"
+)
+
+const (
+	workerKeyDir = ".simulator/keys"
+)
+
+var (
+	transferGasLimit = uint64(21000)
+	transferAmount   = big.NewInt(1)
+	workDelay        = time.Duration(100 * time.Millisecond)
+	retryDelay       = time.Duration(500 * time.Millisecond)
+
+	chainID     *big.Int
+	signer      types.Signer
+	feeCap      *big.Int
+	priorityFee *big.Int
+
+	maxTransferCost  *big.Int
+	requestAmount    *big.Int
+	minFunderBalance *big.Int
+)
+
+func setupVars(cID *big.Int, bFee uint64, pFee uint64) {
+	chainID = cID
+	signer = types.LatestSignerForChainID(chainID)
+	priorityFee = new(big.Int).SetUint64(pFee * params.GWei)
+	feeCap = new(big.Int).Add(new(big.Int).SetUint64(bFee*params.GWei), priorityFee)
+
+	maxTransferCost = new(big.Int).Mul(new(big.Int).SetUint64(transferGasLimit), feeCap)
+	maxTransferCost = new(big.Int).Add(maxTransferCost, transferAmount)
+
+	requestAmount = new(big.Int).Mul(maxTransferCost, big.NewInt(100))
+	minFunderBalance = new(big.Int).Add(maxTransferCost, requestAmount)
+}
+
+func createWorkers(ctx context.Context, keys []*key.Key, endpoints []string, desiredWorkers int) (*worker, []*worker, error) {
+	var master *worker
+	var workers []*worker
+	for i, k := range keys {
+		worker, err := newWorker(k, endpoints[i%len(endpoints)])
+		if err != nil {
+			return nil, nil, err
+		}
+		if err := worker.fetchBalance(ctx); err != nil {
+			return nil, nil, err
+		}
+		if err := worker.fetchNonce(ctx); err != nil {
+			return nil, nil, fmt.Errorf("could not get nonce: %w", err)
+		}
+		log.Printf("loaded worker %s (balance=%s nonce=%d)\n", worker.k.Address.Hex(), worker.balance.String(), worker.nonce)
+
+		switch {
+		case master == nil:
+			master = worker
+		case master.balance.Cmp(worker.balance) < 0:
+			// We swap the worker with the master if it has a larger balance to
+			// ensure the master is the key with the largest balance
+			workers = append(workers, master)
+			master = worker
+		default:
+			workers = append(workers, worker)
+		}
+	}
+
+	var err error
+	if master == nil {
+		master, err = newWorker(nil, endpoints[rand.Intn(len(endpoints))])
+		if err != nil {
+			// This should never happen
+			return nil, nil, fmt.Errorf("unable to create master: %w", err)
+		}
+	}
+
+	for len(workers) < desiredWorkers {
+		i := len(workers)
+		worker, err := newWorker(nil, endpoints[i%len(endpoints)])
+		if err != nil {
+			// This should never happen
+			return nil, nil, fmt.Errorf("unable to create worker: %w", err)
+		}
```

Comment:
can we remove both `this should never happen` comments. This would happen if it's unable to establish a connection to the endpoint.